pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -326,6 +331,14 @@ func (s *NodeController) checkNodeReady(node *api.Node) *api.NodeCondition {
 	}
 }
 
+// sendConditionEvent sends condition change event.
+func (s *NodeController) sendConditionEvent(node *api.Node, kind api.NodeConditionKind, oldCondition, newCondition *api.NodeCondition) {
+	// TODO: if 'oldCondition' is nil, we should publish the initial condition, but selfLink is not set.
+	if oldCondition != nil {
+		record.Eventf(node, "condition change", "Node condition %v changes from %v to %v", kind, oldCondition.Status, newCondition.Status)
```

Comment:
"condition change" is not a good reason; if you look at the docs, a reason should be "machine understandable", as in you should be able to plug it into a switch statement and do something useful with it. The events this makes won't be useful unless you parse the string, which we don't want people doing.